MetaMask/metamask-extension
# Feature: Transaction Insights

### Explanation:

This PR adds the Transactions Insights feature and supporting functionality. This will give users more information about what is happening when sending a transaction to a contract. The core of the functionality is displayed below:

https://user-images.githubusercontent.com/7499938/144235680-024f9aae-586f-4d79-bebc-ad849961d879.mp4

To achieve this, this PR uses https://github.com/trufflesuite/truffle/tree/develop/packages/decoder to decode transaction details which are retrieved from a codefi api

The above screen capture demonstrates the transaction decoding feature and the use of the new modals for viewing and editing account and nickname information.

In addition, the PR adds the transaction decoding to the transaction details modal that is opened when clicking list items in the transaction activity list. This addition also updates the styles/layouts of that modal.

### Manual testing steps:  

This can be done on the Kovan network.

1.   Create a transaction that requires interaction with a contract. Uniswap is a good example
2.   Inspect the "Data" tab to see the transaction details 
3.   Click an address in the transaction details and edit the nickname
4.  Confirm the transaction
5. Click the transaction in the activity list
6. View the transaction information
7. Click "Transaction Data"
8. See the Transaction Data including the nickname you previously edited.

### Screenshots and videos:


https://user-images.githubusercontent.com/7499938/144238081-6a3a2861-25f1-4afa-8ffb-592b1591f087.mp4
